inspirehep/modules/records/validators/rules.py

```diff
@@ -0,0 +1,156 @@
+                                      field='cnum',
+                                      required='document_type',
+                                      values=required_values),
+                                  cause='Error')
+
+
+def check_document_type_if_thesis_info_exist(record):
+
+    if 'thesis_info' in record:
+        required_values = ['thesis']
+        if check_field_values_not_in_required_values_for_record(record, 'document_type', required_values):
+            yield ValidationError(path=[''],
+                                  message=FIELD_REQUIRE_FIELD_VALUES_TEMPLATE.format(
+                                      field='thesis_info',
+                                      required='document_type',
+                                      values=required_values),
+                                  cause='Error')
+
+
+def check_if_isbn_is_valid(record):
+    def _check_isbn_is_valid(field, value, prop, index, errortype):
+        if not is_isbn(value):
+            yield ValidationError(path=['{field}/{index}/{prop}'.format(field=field,
+                                                                        prop=prop,
+                                                                        index=index)],
+                                  message=FIELD_VALIDATION_TEMPLATE.format(
+                                      field='isbns',
+                                      value=value
+                                  ),
+                                  cause=errortype)
+
+    for e in check_if_listfield_property_values_are_valid(record=record,
+                                                          field='isbns',
+                                                          prop='value',
+                                                          checker=_check_isbn_is_valid):
+        yield e
+
+
+def check_if_isbn_exist_in_other_records(record):
+    for e in check_if_listfield_property_values_are_valid(record=record,
+                                                          field='isbns',
+                                                          prop='value',
+                                                          checker=query_db_for_duplicates_for_field_listitem_with_index):
+        yield e
+
+
+def check_if_languages_are_valid_iso(record):
+    def _check_language(language, index):
+        try:
+            pycountry.languages.lookup(language)
+        except LookupError:
+            yield ValidationError(path=['languages/{index}'.format(index=index)],
+                                  message=FIELD_VALIDATION_TEMPLATE.format(
+                                      field='languages',
+                                      value=language
+                                  ),
+                                  cause='Error')
+    for i, item in enumerate(record.get('languages', [])):
+        for e in _check_language(item, i):
+            yield e
```

Comment:
No reason for not having it in the schema as far as I remember.

The discussion was about what languages to present to the user in a dropdown in the user submission, so not related to this.

Then I guess we can create an issue in inspire-schemas. @michamos will have more info once he is back since I think he considered the option of adding it in the schema and maybe discarded it for some reason.